File: app/src/main/java/com/felix/madeclass/model/TvShow.java
package com.felix.madeclass.model;

import android.os.Parcel;
import android.os.Parcelable;

public class TvShow implements Parcelable {
    private int photo;
    private String title;
    private String release_date;
    private String rating;
    private String overview;

    public int getPhoto() {
        return photo;
    }

    public void setPhoto(int photo) {
        this.photo = photo;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getRelease_date() {
        return release_date;
    }

    public void setRelease_date(String release_date) {
        this.release_date = release_date;
    }

    public String getRating() {
        return rating;
    }

    public void setRating(String rating) {
        this.rating = rating;
    }

    public String getOverview() {
        return overview;
    }

    public void setOverview(String overview) {
        this.overview = overview;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeInt(this.photo);
        dest.writeString(this.title);
        dest.writeString(this.release_date);
        dest.writeString(this.rating);
        dest.writeString(this.overview);
    }

    public TvShow() {
    }

    protected TvShow(Parcel in) {
        this.photo = in.readInt();
        this.title = in.readString();
        this.release_date = in.readString();
        this.rating = in.readString();
        this.overview = in.readString();
    }

    public static final Parcelable.Creator<TvShow> CREATOR = new Parcelable.Creator<TvShow>() {
        @Override
        public TvShow createFromParcel(Parcel source) {
            return new TvShow(source);
        }

        @Override
        public TvShow[] newArray(int size) {
            return new TvShow[size];
        }
    };
}

File: app/src/main/java/com/felix/madeclass/model/TvShow.kt
package com.felix.madeclass.model

import android.os.Parcel
import android.os.Parcelable

class TvShow() : Parcelable {

    var tvId: String? = null
    var photoHigh: String? = null
    var photoLow: String? = null
    var photoBackdropHigh: String? = null
    var photoBackdropLow: String? = null
    var title: String? = null
    var releaseDate: String? = null
    var rating: String? = null
    var overview: String? = null
    var episodes: String? = null
    var seasons: String? = null

    constructor(parcel: Parcel) : this() {
        tvId = parcel.readString()
        photoHigh = parcel.readString()
        photoLow = parcel.readString()
        photoBackdropHigh = parcel.readString()
        photoBackdropLow = parcel.readString()
        title = parcel.readString()
        releaseDate = parcel.readString()
        rating = parcel.readString()
        overview = parcel.readString()
        episodes = parcel.readString()
        seasons = parcel.readString()
    }

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeString(tvId)
        parcel.writeString(photoHigh)
        parcel.writeString(photoLow)
        parcel.writeString(photoBackdropHigh)
        parcel.writeString(photoBackdropLow)
        parcel.writeString(title)
        parcel.writeString(releaseDate)
        parcel.writeString(rating)
        parcel.writeString(overview)
        parcel.writeString(episodes)
        parcel.writeString(seasons)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<TvShow> {
        override fun createFromParcel(parcel: Parcel): TvShow {
            return TvShow(parcel)
        }

        override fun newArray(size: Int): Array<TvShow?> {
            return arrayOfNulls(size)
        }
    }


}
